hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -14,13 +15,22 @@ def new_install?
     !File.exist?(success_file)
   end
 
-  def ensure_package(package, state = 'installed')
-    unless ['installed', 'absent', 'latest'].include?(state)
-      fail_and_exit('Incorrect package state supplied to ensure_package')
-    end
+  def ensure_packages(packages, state = 'installed')
+    return if packages.empty?
 
+    code = "package { ['#{packages.join('\', \'')}']: ensure => #{state} }"
     bin_path = Kafo::PuppetCommand.search_puppet_path('puppet')
-    execute("#{bin_path} resource package #{package} ensure=#{state}")
+
+    logger.info("Ensuring #{packages.join(', ')} to package state #{state}")
+    stdout, stderr, stderr, status = Open3.capture3("echo \"#{code}\" | #{bin_path} apply --detailed-exitcodes")
+
+    unless [0, 2].include?(status.exitstatus)
+      log_and_say(:error, "Failed to ensure #{packages.join(', ')} are #{state}")
+      log_and_say(:error, stderr.strip) if stderr && stderr.strip
+      logger.info(stdout.strip) if stdout && stdout.strip
```

Comment:
I still think this is noise that we shouldn't print out. At best, this could be debug information.